lib/Service/OauthService.php

```diff
@@ -58,10 +58,18 @@ public function createNcOauthClient(string $name, string $redirectUri): array {
 		$client->setName($name);
 		$client->setRedirectUri(sprintf($redirectUri, $clientId));
 		$secret = $this->secureRandom->generate(64, self::validChars);
-		if (version_compare(OC_Util::getVersionString(), '27.0.1') >= 0) {
-			$encryptedSecret = $this->crypto->encrypt($secret);
-		} else {
+		if (version_compare(OC_Util::getVersionString(), '30.0.0') >= 0) {
+			$encryptedSecret = bin2hex($this->crypto->calculateHMAC($secret));
+		} elseif (version_compare(OC_Util::getVersionString(), '29.0.7') >= 0 && version_compare(OC_Util::getVersionString(), '30.0.0') < 0) {
+			$encryptedSecret = bin2hex($this->crypto->calculateHMAC($secret));
+		} elseif (version_compare(OC_Util::getVersionString(), '28.0.10') >= 0 && version_compare(OC_Util::getVersionString(), '29.0.0') < 0) {
+			$encryptedSecret = bin2hex($this->crypto->calculateHMAC($secret));
+		} elseif (version_compare(OC_Util::getVersionString(), '27.1.11.8') >= 0 && version_compare(OC_Util::getVersionString(), '28.0.0') < 0) {
+			$encryptedSecret = bin2hex($this->crypto->calculateHMAC($secret));
+		} elseif (version_compare(OC_Util::getVersionString(), '27.0.0') === 0) {
 			$encryptedSecret = $secret;
+		} else {
+			$encryptedSecret = $this->crypto->encrypt($secret);
 		}
```

Comment:
would a switch / case statement look cleaner?
```suggestion
		To convert the `if` statement to a `switch` case statement, we need to use a combination of `true` as the switch expression and `case` statements with conditions. Here is the simplified version using a `switch` statement:

```php
switch (true) {
    case version_compare(OC_Util::getVersionString(), '30.0.0') >= 0:
    case version_compare(OC_Util::getVersionString(), '29.0.7') >= 0 && version_compare(OC_Util::getVersionString(), '30.0.0') < 0:
    case version_compare(OC_Util::getVersionString(), '28.0.10') >= 0 && version_compare(OC_Util::getVersionString(), '29.0.0') < 0:
    case version_compare(OC_Util::getVersionString(), '27.1.11.8') >= 0 && version_compare(OC_Util::getVersionString(), '28.0.0') < 0:
        $encryptedSecret = bin2hex($this->crypto->calculateHMAC($secret));
        break;
    case version_compare(OC_Util::getVersionString(), '27.0.0') === 0:
        $encryptedSecret = $secret;
        break;
    default:
        $encryptedSecret = $this->crypto->encrypt($secret);
        break;
}
```